File: app/src/main/java/com/quranapp/android/utils/univ/RegexPattern.java
package com.quranapp.android.utils.univ;

import java.util.regex.Pattern;

public class RegexPattern {
    public static final Pattern DIACRITICS_PATTERN = Pattern.compile("[^\\p{ASCII}]");
    public static final Pattern DIACRITICS_PATTERN_UNICODE = Pattern.compile("\\p{M}");
    public static final Pattern VERSE_RANGE_PATTERN = Pattern.compile("(\\d+)-(\\d+)");
    public static final Pattern CHAPTER_OR_JUZ_PATTERN = Pattern.compile("(\\d+)");
    public static final Pattern VERSE_JUMP_PATTERN = Pattern.compile("(\\d+)[\\s+]?:[\\s+]?(\\d+)");
    public static final Pattern VERSE_RANGE_JUMP_PATTERN = Pattern.compile("(\\d+)[\\s+]?:[\\s+]?(\\d+)[\\-](\\d+)");
}

File: app/src/main/java/com/quranapp/android/utils/univ/RegexPattern.kt
package com.quranapp.android.utils.univ

import java.util.regex.Pattern

object RegexPattern {
    @JvmField
    val DIACRITICS_PATTERN: Pattern = Pattern.compile("[^\\p{ASCII}]")

    @JvmField
    val DIACRITICS_PATTERN_UNICODE: Pattern = Pattern.compile("\\p{M}")

    @JvmField
    val VERSE_RANGE_PATTERN: Pattern = Pattern.compile("(\\d+)-(\\d+)")

    @JvmField
    val CHAPTER_OR_JUZ_PATTERN: Pattern = Pattern.compile("(\\d+)")

    @JvmField
    val VERSE_JUMP_PATTERN: Pattern = Pattern.compile("(\\d+)[\\s+]?:[\\s+]?(\\d+)")

    @JvmField
    val VERSE_RANGE_JUMP_PATTERN: Pattern = Pattern.compile(
        "(\\d+)[\\s+]?:[\\s+]?(\\d+)[\\-](\\d+)"
    )
}
